Not able to update on current HA installation

ESPHOME will not compile after last update.

it has worked until now, what can cause this problem ?

see below:

src/esphome/components/nibegw/NibeGwComponent.cpp: In member function 'void NibeGwComponent::callback_msg_received(const byte*, int)':
src/esphome/components/nibegw/NibeGwComponent.cpp:33:72: error: invalid cast from type 'std::__tuple_element_t<0, std::tuple<esphome::network::IPAddress, int> >' {aka 'esphome::network::IPAddress'} to type 'uint32_t' {aka 'long unsigned int'}
         if (!udp_read_.writeTo(data, len, (uint32_t)std::get<0>(*target), std::get<1>(*target))) {
                                                                        ^
src/esphome/components/nibegw/NibeGwComponent.cpp: In member function 'void NibeGwComponent::token_request_cache(AsyncUDPPacket&, byte, byte)':
src/esphome/components/nibegw/NibeGwComponent.cpp:59:52: error: conversion from 'uint32_t' {aka 'long unsigned int'} to 'esphome::network::IPAddress' is ambiguous
     network::IPAddress ip = (uint32_t)udp.remoteIP();
                                                    ^
In file included from src/esphome.h:20,
                 from src/esphome/components/nibegw/NibeGwComponent.h:9,
                 from src/esphome/components/nibegw/NibeGwComponent.cpp:1:
src/esphome/components/network/ip_address.h:55:3: note: candidate: 'esphome::network::IPAddress::IPAddress(esp_ip4_addr_t*)' <near match>
   IPAddress(esp_ip4_addr_t *other_ip) { memcpy((void *) &ip_addr_, (void *) other_ip, sizeof(esp_ip4_addr_t)); }
   ^~~~~~~~~
src/esphome/components/network/ip_address.h:55:3: note:   conversion of argument 1 would be ill-formed:
src/esphome/components/nibegw/NibeGwComponent.cpp:59:29: error: invalid conversion from 'uint32_t' {aka 'long unsigned int'} to 'esp_ip4_addr_t*' {aka 'esp_ip4_addr*'} [-fpermissive]
     network::IPAddress ip = (uint32_t)udp.remoteIP();
                             ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from src/esphome.h:20,
                 from src/esphome/components/nibegw/NibeGwComponent.h:9,
                 from src/esphome/components/nibegw/NibeGwComponent.cpp:1:
src/esphome/components/network/ip_address.h:50:3: note: candidate: 'esphome::network::IPAddress::IPAddress(esp_ip6_addr_t*)' <near match>

--- a/components/nibegw/NibeGwComponent.cpp
+++ b/components/nibegw/NibeGwComponent.cpp
@@ -30,7 +30,8 @@ void NibeGwComponent::callback_msg_received(const byte* const data, int len)
     ESP_LOGD(TAG, "UDP Packet with %d bytes to send", len);
     for (auto target = udp_targets_.begin(); target != udp_targets_.end(); target++)
     {
-        if (!udp_read_.writeTo(data, len, (uint32_t)std::get<0>(*target), std::get<1>(*target))) {
+        ip_addr_t address = (ip_addr_t)std::get<0>(*target);
+        if (!udp_read_.writeTo(data, len, &address, std::get<1>(*target))) {
             ESP_LOGW(TAG, "UDP Packet send failed to %s:%d",
                           std::get<0>(*target).str().c_str(),
                           std::get<1>(*target));
@@ -56,8 +57,8 @@ void NibeGwComponent::token_request_cache(AsyncUDPPacket& udp, byte address, byt
         return;
     }
 
-    network::IPAddress ip = (uint32_t)udp.remoteIP();
-    if (udp_source_ip_.size() && udp_source_ip_.count(ip) == 0) {
+    network::IPAddress ip = udp.remoteIP();
+    if (udp_source_ip_.size() && std::count(udp_source_ip_.begin(), udp_source_ip_.end(), ip) == 0) {
         ESP_LOGW(TAG, "UDP Packet wrong ip ignored %s", ip.str().c_str());
         return;
     }

--- a/components/nibegw/NibeGwComponent.h
+++ b/components/nibegw/NibeGwComponent.h
@@ -37,7 +37,7 @@ class NibeGwComponent: public esphome::Component, public esphome::uart::UARTDevi
     const int requests_queue_max = 3;
     int udp_read_port_  = 9999;
     int udp_write_port_ = 10000;
-    std::set<network::IPAddress> udp_source_ip_;
+    std::vector<network::IPAddress> udp_source_ip_;
     bool is_connected_ = false;
 
     std::vector<target_type> udp_targets_;
@@ -67,7 +67,7 @@ class NibeGwComponent: public esphome::Component, public esphome::uart::UARTDevi
     }
 
     void add_source_ip(const network::IPAddress& ip){
-        udp_source_ip_.insert(ip);
+        udp_source_ip_.push_back(ip);
     };
 
     void set_const_request(int address, int token, request_data_type request)

